WireDomain/Tests/WireDomainTests/Event Processing/TeamEventProcessor/TeamMemberLeaveEventProcessorTests.swift

```diff
@@ -0,0 +1,319 @@
+    let modelHelper = ModelHelper()
+
+    var context: NSManagedObjectContext {
+        coreDataStack.syncContext
+    }
+
+    override func setUp() async throws {
+        coreDataStack = try await coreDataStackHelper.createStack()
+        sut = TeamMemberLeaveEventProcessor(
+            context: context
+        )
+        try await super.setUp()
+    }
+
+    override func tearDown() async throws {
+        coreDataStack = nil
+        sut = nil
+        try coreDataStackHelper.cleanupDirectory()
+        try await super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testProcessEvent_It_Deletes_A_Member_When_Receiving_A_Team_Member_Leave_Update_Event_For_Another_User() async throws {
+        // Given
+
+        await context.perform { [context] in
+            let user = ZMUser.insertNewObject(in: context)
+            user.remoteIdentifier = Scaffolding.userID
+
+            let team = Team.insertNewObject(in: context)
+            team.remoteIdentifier = Scaffolding.teamID
+
+            let member = Member.getOrUpdateMember(
+                for: user,
+                in: team,
+                context: context
+            )
+
+            XCTAssertNotNil(member)
+            XCTAssertEqual(user.membership, member)
+        }
+
+        // When
+
+        try await sut.processEvent(Scaffolding.event)
+
+        // Then
+
+        try await context.perform { [context] in
+            XCTAssertNotNil(ZMUser.fetch(with: Scaffolding.userID, in: context))
+
+            /// users won't be deleted as we might be in other (non-team) conversations with them
+            let team = try XCTUnwrap(Team.fetch(with: Scaffolding.teamID, in: context), "No team")
+
+            XCTAssertEqual(team.members, [])
+        }
+    }
+
+    func testProcessEvent_It_Request_Account_Deletion_When_Receiving_A_Team_Member_Leave_Update_Event_For_Self_User() async throws {
+        // Given
+
+        await context.perform { [context] in
+            let user = ZMUser.selfUser(in: context)
+            user.remoteIdentifier = Scaffolding.userID
+
+            let team = Team.insertNewObject(in: context)
+            team.remoteIdentifier = Scaffolding.teamID
+
+            let member = Member.getOrUpdateMember(
+                for: user,
+                in: team,
+                context: context
+            )
+
+            XCTAssertNotNil(member)
+            XCTAssertEqual(user.membership, member)
+        }
+
+        let expectation = XCTestExpectation()
+        let notificationName = AccountDeletedNotification.notificationName
+
+        NotificationCenter.default.addObserver(
+            forName: notificationName,
+            object: nil,
+            queue: nil
+        ) { notification in
+
+            XCTAssertNotNil(notification.userInfo?[notificationName] as? AccountDeletedNotification)
+
+            // Then
+
+            expectation.fulfill()
+        }
+
+        // When
+
+        try await sut.processEvent(Scaffolding.event)
```

Comment:
The expectation is not awaited.

```suggestion
            expectation.fulfill()
        }

        // When

        try await sut.processEvent(Scaffolding.event)

        // Then

        await fulfillment(of: [expectation], timeout: 1)
```

Reply:
fixed, thx